lib/jxl/encode.cc

```diff
@@ -378,6 +378,11 @@ JxlEncoderStatus JxlEncoderAddImageFrame(const JxlEncoderOptions* options,
     return JXL_ENC_ERROR;
   }
 
+  if (options->values.lossless) {
+    // XYB does not make sense for lossless encoding, ensure it is disabled.
+    options->enc->metadata.m.xyb_encoded = false;
```

Comment:
Thank you for proposing the fix. Modifying the value here when adding image frame may not be the correct approach, maybe it could return an error instead here, and JxlEncoderOptionsSetLossless should set this instead. I wonder how this worked before actually!

In which scenario did you encounter this issue? That might help to find the best solution.

Reply:
In the linked issue, you can see the code that generates the issue. The encoder at the moment seems to always want to use XYB encoding, lossless or not.

Reply:
`JxlEncoderOptionsSetLossless` could set it, but in the current code it would be overwritten by `JxlEncoderSetBasicInfo`
This just seemed like the most simple and elegant solution to me at least.

Reply:
I responded to the bug: the solution is to set m_info.uses_original_profile = JXL_TRUE before calling JxlEncoderSetBasicInfo. So lossless, and xyb color are two independent settings, and there's no need to force one in the encoder. Does this fix it?

Reply:
The problem is that xyb_encoded is an image setting (part of SetBasicInfo) so it is too late when adding a frame to change it (since the image header is already written here).

Returning an error when trying to use lossless when xyb_encoded is true would make sense though, since actual lossless is no longer possible in that case.